internal/processor/keys_test.go

```diff
@@ -61,49 +62,366 @@ func TestKeyConversionConvertSpace(t *testing.T) {
 }
 
 func TestKeepKeys(t *testing.T) {
-	keepKeys := []string{"abc", "xyz"}
-	currentSample := map[string]interface{}{
-		"abc":   1,
-		"xyz":   2,
-		"dfadf": "dafsdfa",
-		"def":   "initely busted",
+	getSample := func() map[string]interface{} {
+		return map[string]interface{}{
+			"abc":   1,
+			"xyz":   2,
+			"dfadf": "dafsdfa",
+			"def":   "initely busted",
+		}
 	}
 
-	key := "def"
-
-	RunKeepKeys(keepKeys, &key, &currentSample)
-	output, _ := json.Marshal(currentSample)
-	expected := `{"abc":1,"dfadf":"dafsdfa","xyz":2}`
+	testCases := map[string]struct {
+		keepKeys []string
+		key      string
+		sample   map[string]interface{}
+		expected string
+	}{
+		"WithStrings": {
+			keepKeys: []string{"abc", "xyz"},
+			key:      "def",
+			sample:   getSample(),
+			expected: `{"abc":1,"dfadf":"dafsdfa","xyz":2}`,
```

Comment:
shouldn't the expected only be "abc" and "xyz"?

Reply:
`RunKeepKeys(keepKeys []string, key *string, currentSample *map[string]interface{}) `

this function will remove the  key *string from the map if is not found in the keepKeys []string